linkerd/app/src/identity.rs

```diff
@@ -0,0 +1,97 @@
+use futures::{future, Future};
+pub use linkerd2_app_core::proxy::identity::{
+    certify, Crt, CrtKey, Csr, InvalidName, Key, Local, Name, TokenSource, TrustAnchors,
+};
+use linkerd2_app_core::{
+    classify,
+    config::{ControlAddr, ControlConfig},
+    control, dns, proxy, reconnect,
+    svc::{self, LayerExt},
+    transport::{connect, tls},
+    ControlHttpMetricsRegistry as Metrics, Error, Never,
+};
+use tracing::debug;
+
+#[derive(Clone, Debug)]
+pub enum Config {
+    Disabled,
+    Enabled {
+        control: ControlConfig,
+        certify: certify::Config,
+    },
+}
+
+pub enum Identity {
+    Disabled,
+    Enabled {
+        addr: ControlAddr,
+        local: Local,
+        task: Task,
+    },
+}
+
+pub type Task = Box<dyn Future<Item = (), Error = Never> + Send + 'static>;
```

Comment:
Would it make sense to define this one place instead of multiple times?

Reply:
eh, i don't that all of the modules necessarily have return the same future type (they don't right now, iirc); and it's possible that these could be unboxed and named in individual modules, so i'm -0.